application/defs/include/osib_example.yml

```diff
@@ -0,0 +1,361 @@
+####################################################################################
+# osib_short.yaml: Open Security Information Base (OBIB), (c) OWASP Top 10 team    #
+# This work is licendes under a Creative Commons Attribution 3.0 Unported Licence  #
+# ---------------------------------------------------------------------------------#
+# Example File                                                                     #
+####################################################################################
+
+doctype:        OSIB
+schema:         90         # V 0.90
+date:           20211104
+children:
+  2:
+    alias:      "owasp"
+  "owasp":
```

Comment:
does "owasp" need to be it's own child element?
can it be part of element "2"?
if yes, then do we need this line?
``` "owasp":  ```` 
we already know the element's alias.

Reply:
Yes, I've histirically started with a numbering path for all nodes, as YAML supports names, I'd prefer names as they are more human readable ;-). The alias use case makes more sense for 'aliasing' versions that do not differ in their osib representation. But i's only an extension (a gimmick), no core element of the schema, if you define identical (non breaking) (major) versions and ignore subversions, e.g. asvs V4.0.x -> osib.owasp.asvs.4-0
By the way we sould define a convention how to name versions. For Performance reasons dots ('.') should be avoided, so I use dashes '-' right now.Please finf enclosed the latest OSIB yaml file I use for testing the Python scripts for the two MkDocs macros: https://github.com/sslHello/OSIB-Test/blob/main/osib_short.yml

Reply:
> For Performance reasons dots ('.') should be avoided, 
Can you please elaborate on this? Semantic versioning uses dots

Reply:
> >  Yes, I've historically started with a numbering path for all nodes, as YAML supports names, I'd prefer names as they are more human readable ;-).
> understood. I'll make the "2" be a string then
> **But** I will get rid of the extra "owasp" on line 14 as it duplicates info we already know about
I'd suggest to do just the other way round: use 'owasp' as key for rge ditionary and get rid of the '2', OK?

Reply:
>>   For Performance reasons dots ('.') should be avoided,

> Can you please elaborate on this? Semantic versioning uses dots#

We do need one subtree by version to manage versions. To do this, we add ".<version>" to the path of any context root of any project, standard or tool. This adds "children: { <version>: }" to the tree => adds 1 level of tree depth, assuming that 'children' has not been the sole attribute of the project, standard or tool (e.g. name, source_id etc).
If we use doths inside the version, this works but generates additional layers of depth without any further information:
* Version ".1.0" -> adds "children: {1: { children: { 0: } } }" => 3 layers (= 2 empty layers more than above)
* Version ".1.0.0" -> adds "children: {1: { children: { 0: { children: {0: } } } } }" => 5 layers (= 4 empty layers more than above)
The yaml tree gets blown up and any search/lookup needs to search through these additional layers...
The schema is able to handle both, but a convention *not* to use dots ('.') as seperator in versions produces a taller YAML tree.